bundles/mapping/mapwfs2/domain/WfsLayerModelBuilder.js

```diff
@@ -5,32 +5,85 @@
 Oskari.clazz.define(
     'Oskari.mapframework.bundle.mapwfs2.domain.WfsLayerModelBuilder',
 
-    function (sandbox) {
+    function(sandbox) {
         this.localization = Oskari.getLocalization('MapWfs2');
         this.sandbox = sandbox;
+        this.service = null;
     }, {
+
+        /**
+         * @private @method _checkIfAggregateValuesAreAvailable
+         * function gives value to addLinkToAggregateValues (true/false)
+         *
+         * @return {Boolean}
+         */
+        _checkIfAggregateValuesAreAvailable: function() {
+            this.service = this.sandbox.getService(
+                'Oskari.analysis.bundle.analyse.service.AnalyseService'
+            );
+            if (!this.service) {
+                return false;
+            }
+            return true;
+        },
         /**
          * parses any additional fields to model
          * @param {Oskari.mapframework.domain.WfsLayer} layer partially populated layer
          * @param {Object} mapLayerJson JSON presentation of the layer
          * @param {Oskari.mapframework.service.MapLayerService} maplayerService not really needed here
          */
-        parseLayerData: function (layer, mapLayerJson, maplayerService) {
-            var me = this,
-                toolBuilder = Oskari.clazz.builder('Oskari.mapframework.domain.Tool');
+        parseLayerData: function(layer, mapLayerJson, maplayerService) {
+            var me = this;
 
             if (layer.isLayerOfType("WFS")) {
```

Comment:
This method is called from userlayer, myplaces, analysis modelbuilders. This is why it checks if the layer is WFS even when it's a WFS modelbuilder. While testing that nothing breaks with the changes this could be refactored. Looks like nothing but the default style is set for any non-WFS layertypes.